### PR TITLE
Finished test cases including extra credit, included functions coded …

### PR DESCRIPTION
**Pull request by Stygain**
Includes functions coded by other team members and fulfills extra credit for the assignment with test cases.

### CODE REVIEW

**Comment by parikh1bhavya** [comment]
With this being done we have now the Spanish deck, columns, joker and UML diagrams left to finish.

File: src/main/java/models/Game.java [hunk 1]
```diff
@@ -56,14 +56,14 @@ public void remove(int columnNumber) {
         }
     }
 
-    private boolean columnHasCards(int columnNumber) {
+    boolean columnHasCards(int columnNumber) {
```
Comment (JKiff) [inline src/main/java/models/Game.java]: This causes a violation of encapsulation and information hiding, but is necessary for testing.

File: src/test/models/testCard.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        Card c = new Card(12, Suit.Clubs);
+        c.getSuit();
+    }
+    @Test
+    public void testToString() {
+        Card c = new Card(12, Suit.Clubs);
+        c.toString();
+    }
+
+}
```
Comment (JKiff) [inline src/test/models/testCard.java]: Have edge cases been tested for? For example, attempting to create a card with a value outside the limits. However, this kind of edge case test is not necessary because the code will only ever get called from within.

File: src/test/models/testDeck.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    @Test
+    public void testShuffle() {
+        Deck d = new Deck();
+        d.shuffle();
+    }
+    @Test
+    public void testDealFour() {
+        Deck d = new Deck();
+        d.dealFour();
+    }
```
Comment (JKiff) [inline src/test/models/testDeck.java]: Should these tests include evidence that the functions actually work? for example, printing out the first x cards of the newly dealt or shuffled deck.

File: src/test/models/testColumn.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+import org.junit.Test;
+
+public class testColumn {
+
+    @Test
+    public void testColumnId() {
+        Column c = new Column();
+        Column c2 = new Column(1);
+    }
+
```
Comment (JKiff) [inline src/test/models/testColumn.java]: VARIABLE NAMES: you use c for both card and column in different tests. try using c for card and col for column?

File: src/test/models/testCard.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
+package models;
+
+import org.junit.Test;
+
+public class testCard {
```
Comment (parikh1bhavya) [inline src/test/models/testCard.java]: Make sure make use of comments to make it easier for others looking at your code to understand
